### PR TITLE
feat: add `id` spec

### PR DESCRIPTION
**Pull request by anthonycastaneda**
**What kind of change does this PR introduce? New completion spec for id

**Additional info:**
This is my first spec, and first pull request.  I'm a beginner, so the code may not be correct.  It did pass the npm test and lint:fix.

### CODE REVIEW

File: src/id.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    {
+      name: "-p",
+      description: "Make the output human-readable",
+      exclusiveOn: exclusiveOptions,
+    },
+    {
+      name: "-u",
+      description: "Display the effective user ID as a number",
+      exclusiveOn: exclusiveOptions,
```
Comment (SeparateRecords) [inline src/id.ts:63]: `-n` can only be specified if one of `-G`, `-g`, or `-u` are given

```suggestion
      dependsOn: ["-G", "-g", "-u"],
      exclusiveOn: exclusiveOptions
```

File: src/id.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        name: "-nr",
+        description:
+          "Display the name of the user or group ID for the -G, -g and -u options instead of the number.  If any of the ID numbers cannot be mapped into names the number will be displayed as usual",
+      },
+    },
+  ],
+  // Only uncomment if id takes an argument
+  args: {
+    name: "user",
+    isOptional: true,
```
Comment (SeparateRecords) [inline src/id.ts:74]: This generator _should_ show all the users on the computer! It's lifted [straight out of `killall.ts`](https://github.com/withfig/autocomplete/blob/126dbafce73301ae36af84ee7e163b9923cbf438/src/killall.ts#L116-L124)

```suggestion
    isOptional: true,
    generators: {
      script: "dscl . -list /Users | grep -v '^_'",
      postProcess: (out) =>
        out
          .trim()
          .split("\n")
          .map((username) => ({
            name: username,
            icon: "fig://template?badge=👤",
          })),
    }
```

**Review by SeparateRecords — DISMISSED** [review]
Looking really good!! Left a couple of comments, all super easy to fix up.

There's also one option missing (`-r`) that should be added.

Quick note, don't just use "commit suggestion" since that will do individual commits and make me co-author of them haha

@fedeci I'll defer to you on whether we should keep the `exclusiveOn` or `dependsOn` - it's really hard for me to visualize the relationship between these damn options...

File: src/id.ts (hunk 1 above)
Comment (anthonycastaneda) [reply src/id.ts:63]: The code block shows `-u` but your comment mentions `-n`.  I've made this change on the `-n` option.

File: src/id.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  name: "id",
+  description:
+    "The id utility displays the user and group names and numeric IDs, of the calling process, to the standard output.  If the real and effective IDs are different, both are displayed, otherwise only the real ID is displayed. If a user (login name or user ID) is specified, the user and group IDs of that user are displayed.  In this case, the real and effective IDs are assumed to be the same",
+  options: [
+    {
+      name: "-A",
+      description:
+        "Display the process audit user ID and other process audit properties, which requires privilege",
+      exclusiveOn: exclusiveOptions,
+    },
```
Comment (mschrage) [inline src/id.ts:13]: I'm not sure if you can have "-A" in the list of exclusive options for itself? @fedeci any idea if this is okay or not?

**Review by mschrage — CHANGES_REQUESTED** [review]

File: src/id.ts (hunk 3 above)
Comment (fedeci) [reply src/id.ts:13]: I'm checking!

Comment (fedeci) [reply src/id.ts:13]: It doesn't appear to be a problem!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**